lang/cs/Org.Apache.REEF.Wake/Avro/ProtocolSerializer.cs

```diff
@@ -36,36 +36,52 @@ public sealed class ProtocolSerializer
         private delegate void Deserialize(MemoryStream stream, object observer, long sequence);
 
-        // Message type to serialize/derserialize delagate.
-        private readonly SortedDictionary<string, Serialize> serializeMap = new SortedDictionary<string, Serialize>();
-        private readonly SortedDictionary<string, Deserialize> deserializeMap = new SortedDictionary<string, Deserialize>();
+        /// <summary>
+        /// Map from message type (a string with the message class name) to serializer.
+        /// </summary>
+        private readonly SortedDictionary<string, Serialize>
+            serializeMap = new SortedDictionary<string, Serialize>();
+
+        /// <summary>
+        /// Map from message type (a string with the message class name) to deserializer.
+        /// </summary>
+        private readonly SortedDictionary<string, Deserialize>
+            deserializeMap = new SortedDictionary<string, Deserialize>();
 
         private readonly IAvroSerializer<Header> headerSerializer = AvroSerializer.Create<Header>();
 
         /// <summary>
+        /// Non-generic reflection record for the Register() method of this class. A constant.
+        /// </summary>
+        private static readonly MethodInfo RegisterMethodInfo =
+            typeof(ProtocolSerializer).GetMethod("Register", BindingFlags.Instance | BindingFlags.NonPublic);
+
+        /// <summary>
         /// Register all of the protocol messages using reflection.
         /// </summary>
-        /// <param name="assembly">The Assembley object which contains the namespace of the message classes.</param>
+        /// <param name="assembly">The Assembly object which contains the namespace of the message classes.</param>
         /// <param name="messageNamespace">A string which contains the namespace the protocol messages.</param>
         public ProtocolSerializer(Assembly assembly, string messageNamespace)
         {
-            MethodInfo registerInfo = typeof(ProtocolSerializer).GetMethod("Register", BindingFlags.Instance | BindingFlags.NonPublic);
-            MethodInfo genericInfo;
-
-            Logr.Log(Level.Info, "Retrieving types for assembly: {0}", assembly.FullName);
-            List<Type> types = new List<Type>(assembly.GetTypes());
-            types.Add(typeof(Header));
+            Logr.Log(Level.Verbose, "Retrieving types for assembly: {0}", assembly.FullName);
 
+            var types = new List<Type>(assembly.GetTypes()) { typeof(Header) };
             foreach (Type type in types)
             {
-                string name = type.FullName;
-                if (name.StartsWith(messageNamespace))
+                if (type.FullName.StartsWith(messageNamespace))
                 {
-                    genericInfo = registerInfo.MakeGenericMethod(new[] { type });
+                    MethodInfo genericInfo = RegisterMethodInfo.MakeGenericMethod(new[] { type });
```

Comment:
That's, like, one pointer at the top of the stack. We are talking about one extra `INC RSP` or similar operation in the loop of at most 10 iterations. I can sacrifice that to have a tiny bit more code readability :)